driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -439,7 +456,26 @@ static String escapeId(String ident) {
      * or even {@link Cluster#connect(String)}.
      */
     public static String quote(String id) {
-        return '"' + id + '"';
+        return '"' + id.replace("\"", "\"\"") + '"';
+    }
+
+    /**
+     * Checks whether an identifier is a known reserved CQL keyword or not.
+     * <p/>
+     * The check is case-insensitive, i.e., the word "{@code KeYsPaCe}"
+     * would be considered as a reserved CQL keyword just as "{@code keyspace}".
+     * <p/>
+     * Note: The list of reserved CQL keywords is subject to change in future
+     * versions of Cassandra. As a consequence, this method is provided solely as a
+     * convenience utility and should not be considered as an authoritative
+     * source of truth for checking reserved CQL keywords.
+     *
+     * @param id The identifier to check.
+     * @return {@code true} if the given identifier is a known reserved
+     * CQL keyword, {@code false} otherwise.
+     */
+    public static boolean isReservedCqlKeyword(String id) {
```

Comment:
Small addition to the API, required because it is accessed by `SchemaStatement` (another solution would be to keep duplicate keyword lists).